src/replay/replay_recorder.cpp

```diff
@@ -267,13 +462,96 @@ void ReplayRecorder::save()
                     q->m_suspension_length[1],
                     q->m_suspension_length[2],
                     q->m_suspension_length[3],
+                    q->m_skidding_state,
+                    b->m_attachment,
+                    b->m_nitro_amount,
+                    b->m_item_amount,
+                    b->m_item_type,
+                    b->m_special_value,
+                    r->m_distance,
                     r->m_nitro_usage,
                     (int)r->m_zipper_usage,
-                    r->m_skidding_state,
+                    r->m_skidding_effect,
                     (int)r->m_red_skidding,
                     (int)r->m_jumping
                 );
         }   // for i
     }
     fclose(fd);
 }   // save
+
+/* Returns an encoding value for a given attachment type.
+ * The internal values of the enum for attachments may change if attachments
+ * are introduced, removed or even reordered. To avoid compatibility issues
+ * with previous replay files, we add a layer to encode an enum semantical
+ * value the same way, independently of its internal value.
+ * \param type : the type of attachment to encode */
+
+int ReplayRecorder::enumToCode (Attachment::AttachmentType type)
+{
+    int code =
```

Comment:
All of that could be replaced with (int)type, no? An enum already has an integer value